### PR TITLE
Some nit picks

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const fs = require('fs');
 const jsesc = require('jsesc');
 
 const _template = require('lodash.template');
-const TEST_TEMPLATE = fs.readFileSync('templates/test.case', 'utf8');
+const TEST_TEMPLATE = fs.readFileSync('templates/test.template', 'utf8');
```
Comment (leobalter) [inline build.js:12]: this is a template file, the cases are being loaded by the build.js file

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "unicode-property-escapes-tests",
   "private": true,
   "scripts": {
-    "build": "rm -rf -- output/*.js && node build.js",
+    "build": "rm -rf -- output/ && mkdir output/ && node build.js",
```
Comment (leobalter) [inline package.json:5]: if output/ is not there, it might fail the build.

File: templates/test.template [hunk 3]
```diff
@@ -2,21 +2,20 @@
 // This code is governed by the BSD license found in the LICENSE file.
 
 /*---
-template: default
 author: Mathias Bynens
-desc: Unicode property escapes `\p{<%= UnicodePropertyValueExpression %>}` and `\P{<%= UnicodePropertyValueExpression %>}`
+description: >
+  Unicode property escapes `\p{}` and \P{}` for `<%= UnicodePropertyValueExpression %>`
 esid: sec-static-semantics-unicodematchproperty-p
-flags: [generated]
```
Comment (leobalter) [inline templates/test.template:9]: this is already generating valid test files, this flag would be only used for the original test262 generation tool which does not really apply to these cases.

With this changes, we need to rename desc to description and not reference template anymore.

File: templates/test.template [hunk 4]
```diff
@@ -2,21 +2,20 @@
 // This code is governed by the BSD license found in the LICENSE file.
 
 /*---
-template: default
 author: Mathias Bynens
-desc: Unicode property escapes `\p{<%= UnicodePropertyValueExpression %>}` and `\P{<%= UnicodePropertyValueExpression %>}`
+description: >
+  Unicode property escapes `\p{}` and \P{}` for `<%= UnicodePropertyValueExpression %>`
```
Comment (leobalter) [inline templates/test.template:7]: I also shortened the description here. It's easier - for me - to read. If it's wrong, I can revert this.

File: templates/test.template [hunk 5]
```diff
@@ -2,21 +2,20 @@
...
 esid: sec-static-semantics-unicodematchproperty-p
-flags: [generated]
 features: [regexp-unicode-property-escapes]
 ---*/
 
 assert(
 	/^\p{<%= UnicodePropertyValueExpression %>}+$/u.test(<%= matchSymbols %>),
-	"\\p{<%= UnicodePropertyValueExpression %>} doesn't match all symbols it should match"
+	"\\p{<%= UnicodePropertyValueExpression %>} matches all proper symbols"
```
Comment (leobalter) [inline templates/test.template:14]: in test262 we conventionally use positive messages. We might also use ` x should match all proper symbols`  if it reads better.

**Comment by leobalter** [comment]
The tests are being well generated. One suggestion: I would include the output folder in the git history so we can track changes.

The challenge here is how to tackle the best way to upstream these tests to test262.

I wonder if we could use travis to send a PR to test262 on any update.

File: templates/test.template (hunk 3 above)
Comment (leobalter) [reply templates/test.template:9]: another suggestion, add an info tag to tell which unicode version is being used. This helps as a good reference. E.g:

```yaml
info: |
  Unicode version: <%= UnicodeVersion %>
```

**Comment by mathiasbynens** [comment]
> The challenge here is how to tackle the best way to upstream these tests to test262.

I don’t mind doing that manually — it only happens once a year anyway (once we’ve nailed down the desired test format, that is).

> I wonder if we could use travis to send a PR to test262 on any update.

That would be nice! But might be a lot of work for very little gain. I’ll focus on your other suggestions first.

File: templates/test.template (hunk 3 above)
Comment (mathiasbynens) [reply templates/test.template:9]: > this is already generating valid test files, this flag would be only used for the original test262 generation tool which does not really apply to these cases.

I was wondering about that — thanks for clarifying.

Comment (mathiasbynens) [reply templates/test.template:9]: Shouldn’t we signify somehow that these tests shouldn’t be edited manually? I’ve added a link to this repository in the Info section — maybe that’s enough?

**Pull request merged**